### PR TITLE
Element Isotope t-Student test Module

### PR DESCRIPTION
**Pull request by erickzul**
Based on https://github.com/tnc-br/ddf-isoscapes-api/blob/t-test-corrections/t-test/t_test.ipynb

Note: does not have changes from random sampling that undid TP/FP/TN/FP corrections and t-student test code corrections overall (before git merge error).

### CODE REVIEW

File: hypothesis.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+        # Values from prediction.
+        predicted_isotope_mean = raster.get_data_at_coords(
+            means_isoscape, longitude, latitude, 0)
+        predicted_isotope_variance = raster.get_data_at_coords(
+            variances_isoscape, longitude, latitude, 0)
+        predicted_isotope_sample_count = sample_size_per_location
+
+        # t-student Test
+        _, p_value = scipy.stats.ttest_ind_from_stats(
```
Comment (benwulfe) [inline hypothesis.py:62]: I believe Kader has code that runs the ttest on a single sample -- and we probably want to include that here or at least within ddf_common.  This way, we can be sure that what we eval against is equivalent to production.

**Review by benwulfe — APPROVED** [review]
looks good, but please tell Kader about this code and how it should be updated to call a single invocation of a ttest.

File: hypothesis.py (hunk 1 above)
Comment (erickzul) [reply hypothesis.py]: Sure, per this and offline conversation I've refactored and tested this.

Comment (rothn) [reply hypothesis.py]: I'm still not clear-- where are we calling that code here?

File: hypothesis.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    p_value_threshold: float
+
+
+def sample_ttest(longitude: float,
+                 latitude: float,
+                 isotope_values: pd.Series,
+                 means_isoscape: raster.AmazonGeoTiff,
+                 variances_isoscape: raster.AmazonGeoTiff,
+                 sample_size_per_location: int,
+                 p_value_target: float):
```
Comment (rothn) [inline hypothesis.py:29]: Return type annotation

File: hypothesis.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+    '''
+    longitude: Of the sample
+    latitude: Of the sample
+    isotope_values: Of the sample
```
Comment (rothn) [inline hypothesis.py:33]: Series are ambiguous and can contain multiple fields. Why not a List[float]?

File: hypothesis.py [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+    # Values from prediction.
+    predicted_isotope_mean = raster.get_data_at_coords(
+        means_isoscape, longitude, latitude, 0)
+    predicted_isotope_variance = raster.get_data_at_coords(
+        variances_isoscape, longitude, latitude, 0)
+    predicted_isotope_sample_count = sample_size_per_location
+
+    # t-student Test
+    _, p_value = scipy.stats.ttest_ind_from_stats(
```
Comment (rothn) [inline hypothesis.py:57]: Can we use the `formal-parameter-name = argument-value` syntax here for clarity?

File: hypothesis.py [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+
+    return HypothesisTest(longitude, latitude, p_value, p_value_target)
+
+
+def fraud_metrics(sample_data: pd.DataFrame,
+                  isotope_column_name: str,
+                  means_isoscape: raster.AmazonGeoTiff,
+                  variances_isoscape: raster.AmazonGeoTiff,
+                  sample_size_per_location: int,
+                  p_value_target: float):
```
Comment (rothn) [inline hypothesis.py:75]: Return type annotations

**Pull request merged**